llvm/lib/SandboxIR/SandboxIR.cpp

```diff
@@ -2691,6 +2701,78 @@ PHINode *Context::createPHINode(llvm::PHINode *I) {
   auto NewPtr = std::unique_ptr<PHINode>(new PHINode(I, *this));
   return cast<PHINode>(registerValue(std::move(NewPtr)));
 }
+ICmpInst *Context::createICmpInst(llvm::ICmpInst *I) {
+  auto NewPtr = std::unique_ptr<ICmpInst>(new ICmpInst(I, *this));
+  return cast<ICmpInst>(registerValue(std::move(NewPtr)));
+}
+FCmpInst *Context::createFCmpInst(llvm::FCmpInst *I) {
+  auto NewPtr = std::unique_ptr<FCmpInst>(new FCmpInst(I, *this));
+  return cast<FCmpInst>(registerValue(std::move(NewPtr)));
+}
+
+CmpInst *CmpInst::create(OtherOps Op, Predicate P, Value *S1, Value *S2,
+                         Context &Ctx, const Twine &Name,
+                         Instruction *InsertBefore) {
+  auto &Builder = Ctx.getLLVMIRBuilder();
+  if (InsertBefore)
```

Comment:
I think this `if (InsertBefore)` is not needed. `InsertBefore` should be non-null.